### PR TITLE
Re-added hypervisor annotations

### PR DESCRIPTION
**Pull request by stevenhorsman**
- Add support for setting the sandbox name and namespace in the hypervisor config, which is needed in the remote hypervisor implementation to get the pod name and namespace for the remote pod create request
- Also cherry-pick in the nydus version bump to attempt to stop the dragonball errors

Fixes: #7588

### CODE REVIEW

**Review by fidencio — APPROVED** [review]
lgtm, thanks @stevenhorsman!

**Comment by fidencio** [comment]
And sorry for mistakenly removing it in the first place.

**Comment by stevenhorsman** [comment]
> And sorry for mistakenly removing it in the first place.

It happens to the best of us (and me!) Thanks to @genjuro214 for spotting the issue!

**Comment by fidencio** [comment]
/test

And let's hope the non-triggered tests will be triggered this time

**Comment by stevenhorsman** [comment]
The SEV tests is failing with `not ok 3 [cc][kubernetes][containerd][sev] Test SEV encrypted container launch success with NO measurement` which I think we know is flakey.
@fidencio, @arronwy - if you could check the TDX logs to see if they are real problems and/or re-queue that would be appreciated to try and get peer pods back working. Thanks!

File: src/runtime/virtcontainers/sandbox.go [hunk 1]
```diff
@@ -638,6 +642,23 @@ func newSandbox(ctx context.Context, sandboxConfig SandboxConfig, factory Factor
 	return s, nil
 }
 
+func setHypervisorConfigAnnotations(sandboxConfig *SandboxConfig) {
+	if len(sandboxConfig.Containers) > 0 {
+		// These values are required by remote hypervisor
+		for _, a := range []string{cri.SandboxName, crio.SandboxName} {
```
Comment (wainersm) [inline src/runtime/virtcontainers/sandbox.go:648]: It took me 5 min looking at this line to realize the "cri" and "crio" tokens! hehehe I was like "why the entries are equal?" /me getting old....

**Review by wainersm — APPROVED** [review]
Thanks @stevenhorsman ! This regression really hit me!

**Comment by fidencio** [comment]
/test-tdx

So we can have the result of the QEMU one

**Comment by fidencio** [comment]
Actually, I wouldn't delay it considering everything else's passing.

**Pull request merged**